yt/frontends/arepo/data_structures.py

```diff
@@ -47,14 +47,15 @@ def _is_valid(cls, filename, *args, **kwargs):
         valid = True
         try:
             fh = h5py.File(filename, mode="r")
-            valid = (
+            valid = ((
                 all(ng in fh["/"] for ng in need_groups)
                 and not any(vg in fh["/"] for vg in veto_groups)
                 and (
                     "VORONOI" in fh["/Config"].attrs.keys()
                     or "AMR" in fh["/Config"].attrs.keys()
                 )
-            )
+                # Datasets with GFM_ fields present are AREPO
+                or any(True for field in fh["/PartType0"] if field.startswith("GFM_"))))
```

Comment:
I don't think it does what you think it does. If I'm reading this right it will *always* evaluate to `True`.
I believe _this_ is what you want
```suggestion
                or any(field.startswith("GFM_") for field in fh["/PartType0"])))
```

Reply:
I checked and it seemed to work for both AREPO and non-AREPO gadget datasets.  As far as I can tell, our logic is the same.  But I'm fine using your option because it is more succinct.

Reply:
oh actually you're right, I misread your original implementation. Yeah it would also work, but indeed my solution is more concise and "pythonic", probably. Anyway, glad you like it.